### ISSUE TITLE
Multi-cluster IP range management

### ISSUE THREAD

**Issue by shaase-ctrl** [kubermatic/kubermatic#6421]
**User Story**
As kubermatic user cluster creator,
I have one static IP range for multiple clusters. Each cluster should allocate on his own a dedicate set of IPs (e.g. defined by a parameter input at an KKP addon).

The detailed use case right now ist the output of the configuration of MetalLB, e.g.:
```
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: metallb-system
  name: config
data:
  config: |
    address-pools:
    - name: default
      protocol: layer2
      addresses:
      - {{ .Variables.addressPoolRange }}
```
https://github.com/kubermatic-labs/community-components/blob/master/kubermatic-addons/custom-addon/metallb/address-pools-cm.yaml#L12

In an managed service scenario, the infrastructure setup should  allocate the needed numbers of free virtual IPs by a centralized component or mechanism to avoid conflicting IP allocations for the n used metalLB setups. 

**Acceptance criteria**
- [ ] the solutions provides allocation of a defined number of IPs
- [ ] a release cycle is somehow in place (open for any mechanism)

**Comment by toschneck** [comment]
Additional information regards of the metalLB usage:
- IP manager is in place (Nokia VitalQIP 8.1.2)
- any kind of standard DHCP solution would be great
  - may some DHCP features could be solve this?

Target should be a simple solution, what could may run or get managed at the seed cluster level, for multiple user cluster CIDRs. 
A metalLB issue is also present, but seams to be processing: https://github.com/metallb/metallb/issues/157

Further brainstorming is welcome to find a valuable option for the recent on-premise setups

**Comment by toschneck** [comment]
@rastislavs will create a first proposal

**Comment by rastislavs** [comment]
Proposal: https://github.com/kubermatic/kubermatic/blob/master/docs/proposals/multi-cluster-ipam.md

**Comment by almereyda** [comment]
With [dual-stack](https://kubernetes.io/docs/concepts/services-networking/dual-stack/) becoming the default from Kubernetes 1.21 #6727, is it possible to anticipate changes to this proposal that will allow to natively accommodate IPv6 connectivity?

**Comment by csengerszabo** [comment]
@rastislavs - is this still relevant or can be closed?

**Comment by rastislavs** [comment]
> @rastislavs - is this still relevant or can be closed?

I think this is still relevant. The original proposal may need an update, but the problem of the IPAM across multiple clusters is still valid.

**Comment by rastislavs** [comment]
Implemented.

**Comment by dharapvj** [comment]
@rastislavs - Is this available only in master / 2.21 OR Also we have / we will backport this to 2.19?

**Comment by rastislavs** [comment]
Will be released in KKP 2.21. As this is a completely new feature, it will not be backported to older releases.
